README.md

```diff
@@ -204,10 +206,28 @@ Returns `this` to allow chaining.
 * `def` [`<String>`][] or [`<Number>`][] or [`<Boolean>`][] The default
 
 This sets the default value of the rule. This does not set the actual env
-var though.
+var, but the value in the config's [`<Map>`][] will show the default value if the
+env var's value is empty.
+
+The default value will be applied for an unset var as well as a value of `''`.
 
 Returns `this` to allow chaining.
 
+#### `Definition#allowEmpty()`
+
+This tells the definition to allow the empty string, `''`, as a legitimate value for the
+env var. This empty value will be type casted appropriately for the definition's `type`.
+A common use for this is to allow `''` as a value for a `.string()` definition.
```

Comment:
This is a total nit, but I think it'd be clearer to instead talk about "empty value" for the type, rather than mention empty strings. I think it makes sense when you really think about it (it's env vars, they are always strings) but the mental-model of this package is around the _types_.

Just a suggestion:

> `allowEmpty()` tells the definition to accept an empty value in place of this provided non-empty `.default()`,
for cases where an empty value is valid and/or expected. Without this option, the default value will be used when
an empty value is detected for a given type. A common use for this is to allow `''` as a value for a 
`.string()` definition.

> The empty values for the config types supported by this package are as follows:

Reply:
That's fine.  It always made sense to me to think about strings because the empty string is a prerequisite to ending up with whatever "empty" type.  I'll change this, np.